aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -0,0 +1,641 @@
+        ----------
+        data_shapes:
+            A list of 5- or 6-d tuples
+        data_types:
+            A list of data types
+        dimension_order:
+            The order of dimensions in the data array, using
+            T,C,Z,Y,X and optionally S
+        channel_names:
+            The names for each channel to be put into the OME metadata
+        image_name:
+            The name of the image to be put into the OME metadata
+        pixels_physical_size:
+            X,Y, and Z physical dimensions of each pixel,
+            defaulting to microns
+        channel_colors:
+            The channel colors to be put into the OME metadata
+        is_rgb:
+            is a S dimension present?  S is expected to be the last dim in
+            the data shape
+
+        Returns
+        -------
+        OME
+            An OME object that can be converted to a valid OME-XML string
+        """
+        num_images = len(data_shapes)
+        # resolve defaults that are None
+        if dimension_order is None:
+            dimension_order = [None] * num_images
+        if channel_names is None:
+            channel_names = [None] * num_images
+        if image_name is None:
+            image_name = [None] * num_images
+        if pixels_physical_size is None:
+            pixels_physical_size = [(1.0, 1.0, 1.0)] * num_images
+        if channel_colors is None:
+            channel_colors = [None] * num_images
+
+        # assert all lists are same length
+        if (
+            num_images != len(data_types)
+            or num_images != len(dimension_order)
+            or num_images != len(channel_names)
+            or num_images != len(image_name)
+            or num_images != len(pixels_physical_size)
+            or num_images != len(channel_colors)
+        ):
+            raise ValueError("Mismatched array counts in parameters to build_ome")
+
+        images = []
+        tiff_plane_offset = 0
+        for image_index in range(len(data_shapes)):
+            # correct the dimension_order for ome
+            ome_dimension_order, is_rgb = OmeTiffWriter._resolve_OME_dimension_order(
+                data_shapes[image_index], dimension_order[image_index]
+            )
+            img = OmeTiffWriter._build_ome_image(
+                image_index,
+                tiff_plane_offset,
+                data_shapes[image_index],
+                data_types[image_index],
+                is_rgb,
+                ome_dimension_order,
+                image_name[image_index],
+                pixels_physical_size[image_index],
+                channel_names[image_index],
+                channel_colors[image_index],
+            )
+            # increment tiff_plane_offset for next image
+            tiff_plane_offset += (
+                img.pixels.size_z * img.pixels.size_t * len(img.pixels.channels)
+            )
+            images.append(img)
+
+        ox = OME(creator=f"aicsimageio {get_module_version()}", images=images)
```

Comment:
Just curious, what does `ox` stand for?

Reply:
it's a stupid variable name for ome xml.  I can change it to something better.